editor/plugins/theme_editor_plugin.cpp

```diff
@@ -2949,22 +3135,34 @@ void ThemeTypeEditor::_update_stylebox_from_leading() {
 
 		for (const Ref<StyleBox> &F : styleboxes) {
 			Ref<StyleBox> sb = F;
-			sb->set(E.name, value);
+			ur->add_do_method(*sb, "set", E.name, value);
+			ur->add_undo_method(*sb, "set", E.name, sb->get(E.name));
 		}
 	}
 
+	ur->commit_action();
+
 	leading_stylebox.ref_stylebox = leading_stylebox.stylebox->duplicate();
 
 	// Allow changes to be reported now that the operation is finished.
 	edited_theme->_unfreeze_and_propagate_changes();
 }
 
 void ThemeTypeEditor::_type_variation_changed(const String p_value) {
+	UndoRedo *ur = EditorNode::get_singleton()->get_undo_redo();
+	ur->create_action(TTR("Set Theme Type Variation"));
+
 	if (p_value.is_empty()) {
-		edited_theme->clear_type_variation(edited_type);
+		ur->add_do_method(*edited_theme, "clear_type_variation", edited_type);
 	} else {
-		edited_theme->set_type_variation(edited_type, StringName(p_value));
+		ur->add_do_method(*edited_theme, "set_type_variation", edited_type, StringName(p_value));
 	}
+	ur->add_undo_method(*edited_theme, "set_type_variation", edited_type, edited_theme->get_type_variation_base(edited_type));
```

Comment:
This will err if the current value is empty and you try to use set. You need to use `clear_type_variation` if `edited_theme->get_type_variation_base(edited_type)` returns an empty string.